core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -2886,6 +2886,36 @@ private static boolean isGroupId(AggCall c) {
     return ((AggCallPlus) c).op().kind == SqlKind.GROUP_ID;
   }
 
+  /** Given a list of literals and a target row type, make the literals
+   * respectively match the fields types of the row.
+   *
+   * @param cluster  Cluster
+   * @param rowType  Type expected for values
+   * @param values   A list of literals that should match the rowType */
+  private static List<RexLiteral> convertLiteralTypes(RelOptCluster cluster,
+      RelDataType rowType, List<RexLiteral> values) {
+    assert values.size() == rowType.getFieldCount();
+    RexExecutor executor = cluster.getPlanner().getExecutor();
+    if (executor == null) {
```

Comment:
Do we have a test for the `executor == null`？I'm not sure why this happened.

Reply:
Who knows who is calling the builder and with what setup?
This is just to be safe.

Reply:
+1, getExecutor is a nullable method.

Reply:
Maybe we can improve it by `RexExecutor rexImpl =
        Util.first(cluster.getPlanner().getExecutor(), RexUtil.EXECUTOR);`